lib/lower/convert-expr.cc

```diff
@@ -382,12 +406,10 @@ class ExprLowering {
   /// MIN and MAX operations
   template <Co::TypeCategory TC, int KIND>
   M::Value genval(Ev::Extremum<Ev::Type<TC, KIND>> const &op) {
-    if constexpr (TC == IntegerCat) {
-      return createBinaryFIRTCall<TC, KIND>(
-          op, op.ordering == Ev::Ordering::Greater ? FIRT_MAX : FIRT_MIN);
-    } else {
-      TODO();
-    }
+    std::string name{op.ordering == Ev::Ordering::Greater ? "max" : "min"};
```

Comment:
I'm not sure I like passing arbitrary strings around. They aren't meaningfully type checked by the (C++) compiler, etc.

Reply:
Right, so far intrinsic are not given any kind of type safe tag by the front-end, they are identified only by there name, so I would need to define and map such tag to avoid this here. I think it may make sens to define that in the front-end though.